tests/integration/container_engine.py

```diff
@@ -96,11 +99,140 @@ def run(
         return self._run_cmd(image_cmd)
 
 
+class BuildahEngine(ContainerEngine):
+    """Buildah engine."""
+
+    @property
+    def name(self) -> str:
+        """Get the name of the container engine."""
+        return "buildah"
+
+    @contextmanager
+    def _configure_buildah_container(self, image: str) -> Generator[str, None, None]:
+        """Configure buildah container.
+
+        Buildah requires a container to be explicitly created from an image before it can be used.
+        The built container then can be referenced by the name given when creating it.
+
+        :param image: The image to create a container from.
+        :return: The container name.
+        """
+        container_name = f"hermeto-test-container-{secrets.token_hex(6)}"
+        from_cmd = ["buildah", "from", "--name", container_name, image]
+        _, exit_code = self._run_cmd(from_cmd)
+
+        if exit_code != 0:
+            raise RuntimeError(f"Failed to create buildah container from image {image}.")
+
+        try:
+            yield container_name
+        finally:
+            self._run_cmd([self.name, "rm", container_name])
```

Comment:
What if this fails for some reason? Would it be a hard failure from buildah's POV, should we caught any exception and soft report it without failing a given test?

Reply:
The exception in this case is this `RuntimeError` [here](https://github.com/hermetoproject/hermeto/blob/f9f4410fe73553a8879d1dad170ab6cb76a37d47/tests/integration/container_engine.py#L130). Would you suggest another way to make the tests fail?

Reply:
Ah, that one! Can buildah tell us why it could not create a container, otherwise the error is pretty vague. :)